optimum/habana/transformers/trainer.py

```diff
@@ -992,10 +1003,12 @@ def hpu_deepspeed_checkpointing(function, *checkpoint_args, use_reentrant: Optio
                 remainder = args.gradient_accumulation_steps
             update_step = -1
             total_updates = steps_in_epoch // args.gradient_accumulation_steps + 1
+            if args.gradient_accumulation_steps == 1:
+                total_updates -= 1
             for _ in range(total_updates):
                 update_step += 1
                 num_batches = args.gradient_accumulation_steps if update_step != (total_updates - 1) else remainder
-                batch_samples, num_items_in_batch = self.get_batch_samples_transformers(epoch_iterator, num_batches)
+                batch_samples, num_items_in_batch = self.get_batch_samples(epoch_iterator, num_batches)
```

Comment:
This will probably lead to an error in TRL examples as it is pinned to a quite old version: https://github.com/huggingface/optimum-habana/blob/main/examples/trl/requirements.txt#L1
I'll take a look to see if we can update it.

Reply:
We would need TRL v0.15 to not have a clash between the two `self.get_batch_samples` methods of the trainers, but it requires Accelerate >= 0.34.
My suggestion is to keep `get_batch_samples_transformers` and we can add a TODO saying that this should be removed when the Accelerate dependency is upgraded (which should happen soon, see my comments [above](https://github.com/huggingface/optimum-habana/pull/1794#discussion_r1967806946)).